cmd/kubeadm/app/cmd/init.go

```diff
@@ -589,7 +589,7 @@ func createClient(cfg *kubeadmapi.InitConfiguration, dryRun bool) (clientset.Int
 
 // getDirectoriesToUse returns the (in order) certificates, kubeconfig and Static Pod manifest directories, followed by a possible error
 // This behaves differently when dry-running vs the normal flow
-func getDirectoriesToUse(dryRun bool, defaultPkiDir string) (string, string, string, string, error) {
+func getDirectoriesToUse(dryRun bool, defaultPkiDir string) (certsDirToWriteTo, kubeConfigDir, manifestDir, kubeletDir string, err error) {
```

Comment:
side note: seems that soon we are going to need these in a struct, because multiple phases will need the dirs and all the code from runInit() will be vacated.

Reply:
Agreed. I'll open a ticket to track that work.

Reply:
https://github.com/kubernetes/kubeadm/issues/1177